presto-parquet/src/main/java/com/facebook/presto/parquet/batchreader/decoders/Decoders.java

```diff
@@ -216,6 +223,10 @@ private static ValuesDecoder createValuesDecoder(ColumnDescriptor columnDescript
                         return new Int64TimestampMicrosDeltaBinaryPackedValuesDecoder(valueCount, inputStream);
                     }
 
+                    if (isTimeMicrosType(columnDescriptor)) {
+                        return new Int64TimeMicrosDeltaBinaryPackedValuesDecoder(valueCount, inputStream);
+                    }
+
```

Comment:
Cannot comment above, but we should also support decoder for TimeType when encoding is `RLE_DICTIONARY` or `PLAIN_DICTIONARY`. Parquet will use `PLAIN_DICTIONARY`(v1) or `RLE_DICTIONARY`(v2) as the encoding if it determines that it's worth using dictionary encoding. For example, if we create a table and insert data as follows:

```
CREATE TABLE time_batch_read(a int, b time);
INSERT INTO time_batch_read VALUES (1, time '1:00:00.000'), (2, time '1:00:00.000'), (3, time '1:00:00.000'), (4, time '1:00:00.000'), (5, time '1:00:00.000'), (6, time '1:00:00.000');
```

Parquet will use dictionary encoding for column `b` on writing the data. In this scenario, we will get an error as currently we haven't supported something like `Int64TimeMicrosRLEDictionaryValuesDecoder`.

Reply:
Good catch! I added support and included additional tests in the `presto-parquet` module and updated the test in the Iceberg connector to verify this scenario